sa/_db/migrations/20170423163751_CreateAuthorizationsTable.sql

```diff
@@ -0,0 +1,30 @@
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+CREATE TABLE `authorizations` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `token` varchar(255) NOT NULL,
+  `identifierValue` varchar(255) DEFAULT NULL,
+  `identifierType` int DEFAULT NULL,
+  `registrationID` bigint(20) DEFAULT NULL,
+  `status` int DEFAULT 0,
+  `expires` datetime DEFAULT NULL,
+  `combinations` varchar(255) DEFAULT NULL,
+  `sequence` bigint(20) DEFAULT NULL,
+  `validated` datetime DEFAULT NULL,
+  `validationRecord` mediumblob,
+  `error` mediumblob,
+  `thumbprint` mediumblob,
+
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `token` (`token`),
+  UNIQUE KEY `identifier_key` (`identifierType`, `identifierValue`),
+
+  KEY `a_regId_idx` (`registrationID`) COMMENT 'Common lookup',
```

Comment:
This should be a key on `registrationID, status, expires` because we commonly want to count unexpired pending authzs. And following the convention above, the key name should be the set of field names joined with _.

Also, I think we need a second key on `identifierType, identifierValue, status, registrationID, expires`. This will allow us to efficiently query for the "failed validations" rate limit, *and* make it cross-accounts rather than per-account. This can replace the above "UNIQUE KEY `identifier_key` (`identifierType`, `identifierValue`),"